modules/mono/csharp_script.cpp

```diff
@@ -1052,6 +1056,46 @@ bool CSharpLanguage::debug_break(const String &p_error, bool p_allow_continue) {
 }
 
 #ifdef TOOLS_ENABLED
+void CSharpLanguage::_queue_for_filesystem_update(String p_script_path) {
+	if (!Engine::get_singleton()->is_editor_hint()) {
+		return;
+	}
+
+	if (p_script_path.is_empty()) {
+		return;
+	}
+
+	pending_file_system_update_paths.push_back(p_script_path);
+}
+
+void CSharpLanguage::_flush_filesystem_updates() {
+	if (!Engine::get_singleton()->is_editor_hint()) {
+		return;
+	}
+
+	if (pending_file_system_update_paths.is_empty()) {
+		return;
+	}
+
+	// If the EditorFileSystem singleton is available, update the files;
+	// otherwise, the files will be updated when the singleton becomes available.
+	EditorFileSystem *efs = EditorFileSystem::get_singleton();
+	if (!efs) {
+		return;
+	}
```

Comment:
I think in this case we can clear the `pending_file_system_update_paths` as well, since they those files will also be updated when the EFS singleton becomes available. To prevent updating those files twice.